pkg/render/logstorage/esgateway/esgateway.go

```diff
@@ -0,0 +1,427 @@
+package esgateway
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/dns"
+	"github.com/tigera/operator/pkg/render"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+)
+
+const (
+	EsGatewayName                    = "tigera-es-gateway"
+	EsGatewayServiceAccountName      = "tigera-es-gateway"
+	EsGatewayRole                    = "tigera-es-gateway"
+	EsGatewaySecretKeyName           = "key"
+	EsGatewaySecretCertName          = "tls.crt"
+	EsGatewayVolumeName              = "tigera-es-gateway-certs"
+	EsGatewayElasticServiceName      = "tigera-secure-es-http"
+	EsGatewayKibanaServiceName       = "tigera-secure-kb-http"
+	EsGatewayKibanaPublicCertSecret  = "tigera-secure-kb-http-certs-public"
+	EsGatewayElasticPublicCertSecret = "tigera-secure-es-http-certs-public"
+	EsGatewayElasticUserSecret       = "tigera-es-gateway-elasticsearch-access"
+	EsGatewayTLSSecret               = "tigera-es-gateway-tls"
+	EsGatewayPortName                = "es-gateway-port"
+	EsGatewayPort                    = 5554
+
+	ElasticsearchHTTPSEndpoint = "https://tigera-secure-internal-es-http.tigera-elasticsearch.svc:9200"
```

Comment:
I'm concerned with this Endpoint change. Because the documentation calls out that if providing the `tigera-secure-elasticsearch-cert` that it needs to contain `tigera-secure-es-http.tigera-elasticsearch.svc`, is this PR breaking that?

Reply:
I wasn't aware of this but after taking a look yes, this PR would break that. I can put up a docs PR to change the requirement from `tigera-secure-es-http.tigera-elasticsearch.svc` to `tigera-secure-internal-es-http.tigera-elasticsearch.svc`. Would that be an acceptable solution? Is there a concern that when upgrading a customer would be forced to generate/provide a new cert?

Reply:
Just want to make sure we're on the same page here. 

We're talking about this user doc, right @tmjd: 
https://docs.tigera.io/master/security/comms/log-storage-tls

Basically we give the user the option to provide their own cert/key, instead of our self-signed version, which should be used with the `tigera-secure-es-http.tigera-elasticsearch.svc` domain. 

With the addition of ES Gateway, it's the gateway that has adopted the `tigera-secure-es-http.tigera-elasticsearch.svc` domain. So what should happen here is that the user provided cert/key should be used by ES Gateway, not the Elasticsearch pod. 

The Elasticsearch pod should continue to use the new domain `tigera-secure-internal-es-http.tigera-elasticsearch.svc`. 

Does that make sense to everyone?

Reply:
> With the addition of ES Gateway, it's the gateway that has adopted the tigera-secure-es-http.tigera-elasticsearch.svc domain. So what should happen here is that the user provided cert/key should be used by ES Gateway, not the Elasticsearch pod.

So this means we should allow the user to create/update the `tigera-es-gateway-tls` secret instead of the `tigera-secure-elasticsearch-cert` correct? That would involve making some changes in this PR as well as opening a docs PR to update https://docs.tigera.io/master/security/comms/log-storage-tls.

Reply:
@stevegaossou I think that makes sense, Elasticsearch now becomes strictly internal, so we could just use our own internally generated secrets.

This reminds though that we need to bring @rene-dekker into this so that we don't break certificate management. @rene-dekker could you please review the relevant bits here when you have time?

Reply:
I think what this has made me realize is that `tigera-es-gateway-tls` == `tigera-secure-elasticsearch-cert`. So we don't actually need both of them.

Is there any other use of `tigera-secure-elasticsearch-cert` (referred to as [`TigeraElasticsearchCertSecret` in the Operator code](https://github.com/tigera/operator/search?q=TigeraElasticsearchCertSecret)) other than this? It doesn't look like it. 

Your options are: 

1. Use the existing `tigera-secure-elasticsearch-cert` for ES gateway
- Pro: 
  - This preserves what is already documented to the user and is less confusing.
- Con: 
  - Maybe it's a little less intuitive that ES Gateway's secret is named `tigera-secure-elasticsearch-cert` (although IMO ES Gateway is tied to ES so it's not that bad of a name).

2. Use the new `tigera-es-gateway-tls` for ES gateway and refactor the old refers to `tigera-secure-elasticsearch-cert`
- Pro: 
  - Basically the opposite of the Con for Option 1 (but again I don't think this is a very strong benefit)
- Con: 
  - You have to refactor all the references to `tigera-secure-elasticsearch-cert` in code and user docs
  - User docs change from what they mentioned before (so there's a break in consistency)

My vote is for keeping `tigera-secure-elasticsearch-cert` and getting rid of `tigera-es-gateway-tls`.

Reply:
> My vote is for keeping tigera-secure-elasticsearch-cert and getting rid of tigera-es-gateway-tls.

Mine too, nothing really needs to know about the gateway besides the controller, I don't think we should expose any configuration and mention it as little as possible in places that just care about Elasticsearch (and not an invisible intermediary).

Reply:
> My vote is for keeping tigera-secure-elasticsearch-cert and getting rid of tigera-es-gateway-tls.

I agree with this approach. I'll proceed with the refactor.

Reply:
@rene-dekker you had some comments about the direction we're heading with respect to BYO certs, but it wasn't visible to everyone (namely Erik).

> We currently have no customer at all that uses BYO certs. After much of the certificate management discussions, we were moving in the direction of removing this feature over time altogether, not letting the code BYO them. If they want better certs, we wanna expose the SANs through an API and use certificate management. Therefore I would propose not to allow them at all and use certificate management.

Since we're just going to reuse the `tigera-secure-elasticsearch-cert` cert and keep it consistent, this change shouldn't really push us any further down the BYO cert path. 

Rene, do you have any qualms with Vlad make this PR adopt the cert/key found in `tigera-secure-elasticsearch-cert` for ES Gateway, if a user provides that Secret?

Reply:
Sorry I didn't chime in yesterday, but I think where this has landed is what I think we should do. The gateway changes should require no user facing API changes (even though no one is using it, BYO cert configuration should not change at this time).

As far as dropping the BYO cert capability, that sounds good, but that needs to be explicitly planned work.